koku/koku/middleware.py

```diff
@@ -252,14 +254,15 @@ def _get_tenant_from_db(self, tenant_username, schema_name):
 
         try:
             tenant = Tenant.objects.get(schema_name=schema_name)
-            if schema_name != "public":
-                with KokuTenantMiddleware.tenant_lock:
-                    KokuTenantMiddleware.tenant_cache[tenant_username] = tenant
-                    LOG.debug(f"Tenant added to cache: {tenant_username}")
-            return tenant
         except Tenant.DoesNotExist:
             LOG.info(f"Tenant does not exist: {tenant_username}")
-            return
+            raise Tenant.DoesNotExist()
```

Comment:
Re raise the same exception here rather than raising a new exception to prevent hiding data in the original exception.
```suggestion
            raise
```